build_darwin.sh

```diff
@@ -17,11 +19,17 @@ curl -fsSL "https://github.com/libevent/libevent/releases/download/release-$LIBE
 #they forgot to decorate them with appropriate AVAILABLE_MAC_OS_VERSION checks. 
 #So we have to explicitly disable them for binaries to work on MacOS 10.11.
 
+# Using test/regress.c from a local repo because make check hangs in libevent
+# See: https://github.com/libevent/libevent/issues/747 for more details
+# Updated test/regress.cc disables: del_wait | immediatesignal | 
+# signal_switchbase | signal_while_processing tests
+
 gpg --keyserver "$KEYSERVER" --recv-keys $LIBEVENT_KEY && \
 gpg libevent-$LIBEVENT_VERSION.tar.gz.asc && \
 echo "$LIBEVENT_HASH  libevent-$LIBEVENT_VERSION.tar.gz" | shasum -a 256 -c - && \
 tar -zxvf libevent-$LIBEVENT_VERSION.tar.gz && \
 cd libevent-$LIBEVENT_VERSION && \
+curl -o test/regress.c https://raw.githubusercontent.com/jumde/libevent/legacy_fork_failure/test/regress.c && \
```

Comment:
Issue to revert this change once the upstream change is fixed: https://github.com/brave/tor_build_scripts/issues/34

Reply:
Can we store this file in the repository?  If not, let's at least commit a hash of the file and verify it before using it.